File: examples/C/Data_Management_vector_sum_exercise.c
#include <stdio.h>
#include <stdlib.h>
int main(void){
    int size = 10000;
    int a[size], b[size], c[size];

    // Insert OpenACC directive
    for(int i=0; i<size; ++i){
        a[i] = i;
        b[i] = 2*i;
    }

    // Insert OpenACC directive
    for (int i=0; i<size; ++i){
        c[i] = a[i]+b[i];
    }

    printf("value at position 14: %d\n", c[14]); 
}

File: examples/Fortran/Data_Management_vector_sum_exercise.f90
program vector_sum
    use iso_fortran_env, only : INT32, REAL64
    use openacc
    implicit none

    integer(kind=INT32), parameter              :: system_size  = 10000
    integer(kind=INT32), dimension(system_size) :: a, b, c
    integer(kind=INT32)                         :: i

    ! Insert OpenACC directive
    do i = 1, system_size
        a(i) = i
        b(i) = i * 2
    enddo

    ! Insert OpenACC directive
    do i = 1, system_size
       c(i) = a(i) + b(i)
    enddo

    write(0,"(a22,i3)") "value at position 12: ", c(12)

end program vector_sum        
